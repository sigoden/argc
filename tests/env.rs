use rstest::rstest;

#[rstest]
fn env_help() {
    snapshot_env!(args: ["-h"], envs: {});
}

#[rstest]
fn env_help_subcmd() {
    snapshot_env!(args: ["run", "-h"], envs: {});
}

#[rstest]
fn env_missing() {
    snapshot_env!(args: [], envs: {});
}

#[rstest]
fn env_choice() {
    snapshot_env!(args: [], envs: {"TEST_EB": "1", "TEST_ECA": "val"});
}

#[rstest]
fn env_choice_fn() {
    snapshot_env!(args: [], envs: {"TEST_EB": "1", "TEST_EFA": "val"});
}

#[rstest]
fn env_run() {
    snapshot_env!(args: [], envs: {"TEST_EB": "1"});
}
