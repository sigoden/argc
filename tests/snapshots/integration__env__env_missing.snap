---
source: tests/env.rs
expression: "format!(r#\"\n# OUTPUT\n{output}\n\n# BUILD_OUTPUT\n{build_output}\n\"#)"
---

# OUTPUT
error: the following required environments were not provided:
  TEST_EB


# BUILD_OUTPUT
error: the following required environments were not provided:
  TEST_EB


